src/App.jsx

```diff
@@ -14,33 +15,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goodsList, setGoodsList] = useState([...goodsFromServer]);
+  const [sortType, setSortType] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const applySorting = (type, reversed = false) => {
+    const sorted = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    switch (type) {
+      case 'alphabet':
+        sorted.sort((a, b) => a.localeCompare(b));
+        break;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case 'length':
+        sorted.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+
+    if (reversed) {
+      sorted.reverse();
+    }
+
+    setGoodsList(sorted);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabet' ? '' : ' is-light'}`}
+          onClick={() => {
+            setSortType('alphabet');
+            applySorting('alphabet', isReversed);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'length' ? '' : ' is-light'}`}
+          onClick={() => {
+            setSortType('length');
+            applySorting('length', isReversed);
+          }}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-info ${isReversed ? '' : ' is-light'}`}
+          onClick={() => {
+            const newReversed = !isReversed;
+
+            setIsReversed(newReversed);
+
+            if (sortType === null) {
+              const base = [...goodsFromServer];
+
+              setGoodsList(newReversed ? base.reverse() : base);
+            } else {
+              applySorting(sortType, newReversed);
+            }
+          }}
```

Comment:
no no If you have logic - should be handler , move to const, but in this case you should only change state